agent/consul/state/intention_test.go

```diff
@@ -1199,6 +1199,131 @@ func TestStore_IntentionsList(t *testing.T) {
 	})
 }
 
+func TestStore_IntentionMatch_ConfigEntries(t *testing.T) {
```

Comment:
At first I tried to re-jig `TestStore_IntentionMatch_table` to handle peers but found it a bit too rigid to work with, since it depends on reversing inputs to test source/destination match queries.

Hopefully this test can be the new matrix test after we remove legacy intentions.